src/foundry/common/utils/helpers.mjs.d.ts

```diff
@@ -284,7 +284,9 @@ type RemoveDeletingObjectKeys<T> = RemoveNever<
   }
 >;
 
-type MergeObjectProperty<T, U, M extends MergeObjectOptions> = T extends Record<string, any>
+type MergeObjectProperty<T, U, M extends MergeObjectOptions> = T extends Array<any>
+  ? U
```

Comment:
I'm not sure this is entirely correct, merging `["foo"]` and `[2]` should result in `["foo", 2]` and I think this creates a type signature of `number[]`.
I can't check right now but I think `insertKeys` will cause different behaviour too?

Reply:
I tested it in foundry and for most cases the implementation was correct. But `enforceTypes` allows to change the type from `string[]` to `number[]`, so I handled `enforceTypes` a bit difference.

![image](https://user-images.githubusercontent.com/571106/123739458-fd410680-d8a6-11eb-83bf-ef1c76834088.png)

Reply:
Huh, I was sure it merged arrays too. Well, thanks for looking